### PR TITLE
makefile:refactor docker Makefile

### PR DESCRIPTION
**Pull request by nopeslide**
Refactoring of the docker Makefile

### CODE REVIEW

**Review by pascalharp — APPROVED** [review]
lgtm

File: docker/Makefile [hunk 1]
```diff
@@ -169,154 +132,118 @@ code vscode: | docker-vscode check-docker
...
-			echo make install-docker-fedora; \
-			echo; \
-		else \
-			echo UNKNOWN; \
-			echo please run your package manager to install docker; \
-		fi; \
-		) 1>&2; \
-		exit 1; \
-	fi
+check-docker-installed:
```
Comment (ikstream) [inline docker/Makefile:185]: I am still not convinced we should check for requirements on the software side, but wont block this PR based on it.

**Review by ikstream — COMMENTED** [review]
So far lgtm, will try to test touched features now

File: docker/Makefile (hunk 1 above)
Comment (nopeslide) [reply docker/Makefile:185]: this is sth we have to discuss outside of this refactor PR

**Pull request merged**